assets/src/edit-story/components/panels/publish/index.js

```diff
@@ -0,0 +1,246 @@
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.86)};
+`;
+
+const Time = styled.span`
+  color: ${({ theme }) => rgba(theme.colors.fg.v1, 0.4)};
+`;
+
+const DateTimeWrapper = styled.div`
+  position: relative;
+`;
+
+const StyledToggleIcon = styled(ToggleIcon)`
+  height: 26px;
+  flex: 1;
+`;
+
+function PublishPanel() {
+  const {
+    state: { users },
+  } = useInspector();
+
+  const {
+    state: {
+      meta: { isSaving },
+      story: { author, date, featuredMediaUrl, publisherLogoUrl },
+      capabilities,
+    },
+    actions: { updateStory },
+  } = useStory();
+
+  const [showDatePicker, setShowDatePicker] = useState(false);
+  const dateTimePickerNode = useRef();
+  const dateFieldNode = useRef();
+
+  useKeyDownEffect(
+    dateFieldNode,
+    { key: ['space', 'enter'] },
+    () => {
+      setShowDatePicker(!showDatePicker);
+    },
+    [showDatePicker]
+  );
+
+  useFocusOut(dateTimePickerNode, () => setShowDatePicker(false), [
+    showDatePicker,
+  ]);
+
+  const handleDateChange = useCallback(
+    (value, close = false) => {
+      if (close && showDatePicker) {
+        setShowDatePicker(false);
+      }
+      updateStory({ properties: { date: value } });
+    },
+    [showDatePicker, updateStory]
+  );
+
+  const handleChangeCover = useCallback(
+    (image) =>
+      updateStory({
+        properties: {
+          featuredMedia: image.id,
+          featuredMediaUrl: image.sizes?.medium?.url || image.url,
+        },
+      }),
+    [updateStory]
+  );
+
+  // @todo Enforce square image while selecting in Media Library.
+  const handleChangePublisherLogo = useCallback(
+    (image) => {
+      updateStory({
+        properties: {
+          publisherLogo: image.id,
+          publisherLogoUrl: image.sizes?.thumbnail?.url || image.url,
+        },
+      });
+    },
+    [updateStory]
+  );
+
+  const handleChangeValue = useCallback(
+    (prop) => (value) => updateStory({ properties: { [prop]: value } }),
+    [updateStory]
+  );
+
+  const authorLabel = __('Author', 'web-stories');
+  return (
+    <SimplePanel name="publishing" title={__('Publishing', 'web-stories')}>
+      <Row>
+        <FieldLabel>{__('Publish', 'web_stories')}</FieldLabel>
+        <BoxedText
+          aria-pressed={showDatePicker}
+          aria-haspopup={true}
+          aria-expanded={showDatePicker}
+          onClick={(e) => {
+            e.preventDefault();
+            if (!showDatePicker) {
+              // Handle only opening the datepicker since onFocusOut deals with closing.
+              setShowDatePicker(true);
+            }
+          }}
+          ref={dateFieldNode}
+        >
+          <DateWrapper>
+            <Date>{getReadableDate(date)}</Date>{' '}
+            <Time>{getReadableTime(date)}</Time>
+          </DateWrapper>
+          <StyledToggleIcon />
+        </BoxedText>
+        {showDatePicker && (
+          <DateTimeWrapper>
+            {/* @todo get the actual value for is12Hour */}
```

Comment:
For reference, this is what Gutenberg does:

https://github.com/WordPress/gutenberg/blob/f63053cace3c02e284f00918e1854284c85b9132/packages/editor/src/components/post-schedule/index.js#L10-L20

(with explanation: https://github.com/WordPress/gutenberg/blob/c0e4f538ad2ed96eed064b39657b35f5645238a1/packages/components/src/date-time/README.md#usage)

Since WordPress already localizes Moment.js, we might be able to get the date time setting from there.

Reply:
Or we should just pass all the date settings in the config?

Reply:
Added using the `time_format` option for detecting the time format. 8cbf880